crates/database/src/controller/users.rs

```diff
@@ -64,8 +84,22 @@ impl DbController for UserDatabaseController {
 
     #[instrument(name = "charted.db.users.get_nos", skip(self))]
     async fn get_by_nos(&self, nos: NameOrSnowflake) -> Result<Option<Self::Entity>> {
+        let mut cache = self.worker.lock().await;
+
         match nos {
-            NameOrSnowflake::Snowflake(uid) => self.get(uid).await,
+            NameOrSnowflake::Snowflake(uid) => {
+                let key = CacheKey::user(uid.try_into().unwrap());
+                if let Some(cached) = cache.get(key).await? {
+                    return Ok(Some(cached));
+                }
+
+                self.get(uid).await
+            }
+
+            // we can't do cache calucations on Names since we don't need
+            // duplication, but maybe a "pointer" to their ID?
+            //
+            // TODO(@auguwu): determine if pointers (point name -> id) to a resource is acceptable
```

Comment:
Would it be worth splitting this into an issue so we can track it as the design evolves?

Reply:
Yeah, I think an issue can be created about this